api/log/log_test.go

```diff
@@ -0,0 +1,83 @@
+package log_test
+
+import (
+	"bufio"
+	"net"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/globocom/glbgelf"
+	apiContext "github.com/globocom/huskyCI/api/context"
+
+	"github.com/globocom/huskyCI/api/log"
+)
+
+func startGelfServer(t *testing.T, wg *sync.WaitGroup, msgCount int) {
+	defer wg.Done()
+
+	// set apiContext vars
+	listener, err := net.Listen("tcp", ":12201")
+	if err != nil {
+		t.Errorf("expected to listening server, but failed: %s", err)
+		return
+	}
+	// We could use ExampleInitLog, however, I think it is better if we read it from the stream.
+	conn, err := listener.Accept()
+	if err != nil {
+		t.Errorf("could not accept messages: %s", err)
+		return
+	}
+	var gotMsgs int
+	conn.SetReadDeadline(time.Now().Add(3 * time.Second))
+	reader := bufio.NewReader(conn)
+	for {
+		b, err := reader.ReadByte()
+		if err != nil {
+			t.Errorf("could not read bytes from conn: %s", err)
+			return
+		}
+		if b == '\u0000' { // as per gelf docs: a message is finished when it is NUL terminated
+			gotMsgs++
+		}
+		if gotMsgs == msgCount {
+			break
+		}
+	}
+	if gotMsgs != msgCount {
```

Comment:
This condition contradicts the one in line 43. It will never happen and tests will always pass.

Reply:
Oh yes, thats true! I did not notice that it doesn't make sense *at all*! Sorry for this one, I'll fix it asap.